e2e/tests/query.feature

```diff
@@ -0,0 +1,391 @@
+	And the pocket client should have exited without error
+
+  Scenario: To query the list of params
+	Given the user is running the network "single_node_network"
+	And the user has a pocket client
+	When the user runs the command "query params"
+	Then the user should be able to see standard output containing "param_key"
+	And the user should be able to see standard output containing "param_value"
+	And the pocket client should have exited without error
+
+  Scenario: To query the list of pocketcore params
+	Given the user is running the network "single_node_network"
+	And the user has a pocket client
+	When the user runs the command "query pocket-params"
+	Then the user should be able to see standard output containing "claim_expiration"
+	And the pocket client should have exited without error
+
+  Scenario: To query the list of pocketcore params, Return error with incorrect height
+	Given the user is running the network "single_node_network"
+	And the user has a pocket client
+	When the user runs the command "query pocket-params 9999999"
+	Then the user should be able to see standard output containing "error loading store"
+	And the pocket client should have exited without error
+
+  Scenario: To query a specific node's signing info
+	Given the user is running the network "single_node_network"
+	And the user has a pocket client
+	When the user runs the command "query signing-info {{address}} 0" against validator 0
+	Then the user should be able to see standard output containing "/v1/query/signinginfo"
+	And the user should be able to see standard output containing "result"
+	And the pocket client should have exited without error
+
+  Scenario: To get supplies
+	Given the user is running the network "single_node_network"
+	And the user has a pocket client
+	When the user runs the command "query supply"
+	Then the user should be able to see standard output containing "/v1/query/supply"
+	And the user should be able to see standard output containing "app_staked"
+	And the pocket client should have exited without error
+
+  Scenario: To get supplies, Returns error code with wrong height
+	Given the user is running the network "single_node_network"
+	And the user has a pocket client
+	When the user runs the command "query supply 999999"
+	Then the user should be able to see standard output containing "error loading store"
+	And the pocket client should have exited without error
+
+  Scenario: To get existing supported networks
+	Given the user is running the network "single_node_network"
+	And the user has a pocket client
+	When the user runs the command "query supported-networks"
+	Then the user should be able to see standard output containing "/v1/query/supportedchains"
+	And the user should be able to see standard output containing "0011"
+	And the pocket client should have exited without error
+
+  Scenario: To get existing supported networks, Returns an error code with invalid height
```

Comment:
Confusing scenario. What is the acceptance criterion here?

Reply:
@AneudyM I completely with you that this scenario, and it's title are confusing. When reviewing code, I usually try to use Github's "suggestion" feature so the reviewee can just accept/commit it or modify it if need be.

For example (I have no clue if this is clearer):
```suggestion
  Scenario: To verify that an invalid height cannot be quried from the list of supported-networks
```

Re the `scenario` - 
1. Are we verifying we can't query an invalid height OR we can't query an unsupported network?

Re the `Then` - `error loading store` makes me think the network is not supported.
2. Is that correct? 
3. If so, is there any way to make the source code (without comments) make that more explicit?

Re the `And`
4. Does this exit successfully because the client worked but the backing data is invalid?

Reply:
Yeah. Those are good points. Mmm..., looking at the validation of the scenario it might be trying to query a network with an invalid height. In that case, I think I would rewrite it as follows:

````
Scenario: Return error when querying supported networks with invalid height number
  Given a “single_node_network”
    And a valid pocket client
  When querying a supported network with invalid height “99999” number
  Then an “error loading store” message should be returned
    But the pocket client should not exit with an error status
````

Re point 4: 
That's why I am unsure of including the exit status steps here. What does the specification say about these exit statuses.